velox/common/memory/Memory.cpp

```diff
@@ -58,6 +58,12 @@ MemoryManager::MemoryManager(const MemoryManagerOptions& options)
               .checkUsageLeak = options.checkUsageLeak,
               .debugEnabled = options.debugEnabled})} {
   VELOX_CHECK_NOT_NULL(allocator_);
+  if (arbitrator_ != nullptr) {
+    VELOX_CHECK_EQ(
```

Comment:
Could you add an invalid test case here in MemoryManagerTest with a fake factory method? Thanks!